lua/lvim/core/builtins/init.lua

```diff
@@ -23,12 +23,28 @@ local builtins = {
   "lvim.core.mason",
 }
 
+---@class LvimBuiltin
+---@field active boolean is builtin enabled
+---@field on_config_done function function called after the builtin is set up
+
 function M.config(config)
+  ---@type {[string]: LvimBuiltin}
+  lvim.builtin = {}
+
   for _, builtin_path in ipairs(builtins) do
     local builtin = reload(builtin_path)
 
     builtin.config(config)
   end
 end
 
+function M.setup_builtin(modname)
+  local plugin = require("lvim.core." .. modname).setup()
+
+  local builtin_tbl = lvim.builtin[string.gsub(modname, "-", "_")]
+  if type(builtin_tbl.on_config_done) == "function" then
```

Comment:
you're not checking if `setup` has returned correctly

```suggestion
  if plugin and type(builtin_tbl.on_config_done) == "function" then
```

Reply:
not all builtins return a plugin (commands for example)

Reply:
that's why I mentioned that older PR to cleanly separate builtins from core, you could also check the table at the top of this file.